### PR TITLE
Tempo: Add support for /api/v2/traces/ and partial traces in Trace View

### LINKED ISSUE #92464: Tempo: Add support for new /api/v2/traces/ endpoint
**Why is this needed**:

The existing ` /api/traces ` endpoint has two limitations:
- It's not [OTEL](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/trace/v1/trace.proto) compatible
- When a trace exceeds the maximum size, an error is returned, which can be frustrating for users trying to identify which services are generating these large traces.

**What would you like to be added**:

A new ` /api/v2/traces ` endpoint has been created to address some issues of the current version:
- Be fully OTEL compatible 
- Support returning partial traces

When a trace exceeds the maximum size it will return the following:

```json
{
  "trace": {
      "resourceSpans": [
      ]
  },
  "status": "PARTIAL",
  "message": "Trace exceeds maximum size of 5 bytes, a partial trace is returned"
}
```
**Be aware of the "resourceSpans" that substitutes the previous "batches"**

You can check the message definition here:
https://github.com/grafana/tempo/blob/496c132883c700d67b06573d6f4488486e077773/pkg/tempopb/tempo.proto#L58

PR here: https://github.com/grafana/tempo/pull/3941

### PR DESCRIPTION
**Pull request by adrapereira**
**What is this feature?**

Adds support for the `/api/v2/traces/` API from Tempo added in https://github.com/grafana/tempo/pull/3941 which allows Tempo to return partial traces instead of throwing an error when the trace exceeds the configured limit. 

Grafana will try to query all traces by ID using the `v2` endpoint and will fallback to the `v1` endpoint if it receives a `404` response. This will be useful to support versions of Tempo older than Tempo 2.6 which didn't support `v2`.

The Trace View has also been updated to support showing a badge for partial traces, with a message on a tooltip explaining why the trace was only partially returned. 

image

**Why do we need this feature?**

This feature will help reduce errors when querying large traces. 

**Who is this feature for?**

Users of the Tempo datasource. 

**Which issue(s) does this PR fix?**:

Fixes #92464

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

**Comment by knylander-grafana** [comment]
Do we need to update the docs for this change?

**Review by joey-grafana — CHANGES_REQUESTED** [review]
Pretty cool feature, thank you for opening this PR 👍 

What's the best way to test this in the UI? As in, trigger a partial trace?

I've noticed a crash for null tags `[null, null, ..]` but this only happens when `externalCorePlugins` is not enabled for Tempo. I've tested this against main and it only seems to happen on this branch. 

Screenshot 2024-10-16 at 09 38 32
Screenshot 2024-10-16 at 09 38 52
Screenshot 2024-10-16 at 09 42 30


Also, we updated a few core modules that might effect other parts of Grafana? I'd suggest a look from the backend squad or similar for the go.mod/go.sum changes before merging.

I had to go get a few packages to remove errors in trace_transform.go and trace.go. You will prob need to update go.sum

Screenshot 2024-10-15 at 16 53 06

**Review by joey-grafana — APPROVED** [review]
It appears to be gone as/but I have not found any traces with null tags. I would also add a check in the frontend for null tags/other nulls where the crash is caused in the stack trace above before merging. 

I think it would be no harm to add a 'How to test' section to the description of this PR so it's abundantly clear :) 

I'm going to hit the approve here so you're not blocked when I'm away but there are still some failing tests so please use your best judgement and feel free to ask for a second reviewer if necessary. Note I didn't get a chance to setup the auth in a proper env to see the partial traces notice in the UI but the code generally LGTM 👍

**Pull request merged**
